nodedb.go

```diff
@@ -535,28 +548,42 @@ func (ndb *nodeDB) cacheNode(node *Node) {
 }
 
 // Write to disk and memDB
-func (ndb *nodeDB) Commit() {
+func (ndb *nodeDB) Commit() error {
 	ndb.mtx.Lock()
 	defer ndb.mtx.Unlock()
 
+	var err error
 	if ndb.opts.KeepEvery != 0 {
 		if ndb.opts.Sync {
-			ndb.snapshotBatch.WriteSync()
+			err = ndb.snapshotBatch.WriteSync()
+			if err != nil {
+				return err
+			}
 		} else {
-			ndb.snapshotBatch.Write()
+			err = ndb.snapshotBatch.Write()
+			if err != nil {
+				return err
+			}
 		}
 		ndb.snapshotBatch.Close()
 	}
 	if ndb.opts.KeepRecent != 0 {
 		if ndb.opts.Sync {
-			ndb.recentBatch.WriteSync()
+			err = ndb.recentBatch.WriteSync()
+			if err != nil {
+				return err
```

Comment:
If there's an error here should we delete the write in snapshotDB? Could be ok with either behavior. Just want to see if there's been thought on this

Pro of deleting:
Maintains invariant that pruning logic is always correct

Pros of not deleting:
Simpler
Doesn't seem like it would create obvious issues if we don't delete

Reply:
I haven't thought about it. i think it would be your call as you are the main person for this repo @AdityaSripal

Reply:
I think attempting to delete to maintain state would be ideal, since down the road it will be harder to tell in what situations can we run into states that we don't anticipate in.

Reply:
The previous logic was that if there is an error it would panic. If there is an error here then there is a larger problem at hand and the application would most likely panic. I do believe this is how the sdk will be handling it. I can add documentation for this in the godoc.

Reply:
Perhaps it may be worth not deleting here, since the Delete function may return another error. It's unclear what to do in this case. And it still makes it possible for the invariant to be broken

I think we just return wrapped errors so the user at least knows where the execution failed and can determine DB state from this information

Reply:
So im for leaving it as it is